code/modules/mob/living/carbon/human/species/kidan.dm

```diff
@@ -43,7 +43,161 @@
+	holder.icon = icon
+	holder.icon_state = icon_state
+	holder.alpha = 220
+
+	// Delete itself after some time if it is not permanent variant
+	if(lifespan)
+		QDEL_IN(src, lifespan)
+
+/obj/effect/kidan_pheromones/examine(mob/user)
+	. = ..()
+	if(encoded_message)
+		. += "It has the following message: \"[encoded_message]\""
+	// Failsafe for mappers/adminspawns if they forgot to add a message
+	else
+		. += "Its meaning is incomprehensible."
+
+// For mappers/adminspawns, this one does not self-delete
+/obj/effect/kidan_pheromones/permanent
+	lifespan = null
+
+// Innate action for creating pheromones and destroying current ones, owned by all kida
+/datum/action/innate/produce_pheromones
+	name = "Produce Pheromones"
+	check_flags = AB_CHECK_CONSCIOUS
+	icon_icon = 'icons/effects/effects.dmi'
+	button_icon_state = "kidan_pheromones_static"
+
+	/// How long our message can be (characters)
+	var/maximum_message_length = 30
+
+	/// How many active pheromones we can have
+	var/active_pheromones_maximum = 3
+
+	/// Which currently existing pheromones belong to us
+	var/list/active_pheromones_current
+
+/datum/action/innate/produce_pheromones/Activate()
+	var/mob/living/carbon/human/H = owner
+
+	// Do we want to make or destroy them?
+	switch(alert(H, "Would you like to produce or destroy nearby pheromones?", "Produce Pheromones", "Produce", "Destroy"))
+		// We look for nearby pheromones, if they belong to us, we can destroy them
+		if("Destroy")
+			var/obj/effect/kidan_pheromones/pheromones_to_destroy = locate(/obj/effect/kidan_pheromones) in range(1, H)
+			// No pheromones nearby
+			if(!pheromones_to_destroy)
+				to_chat(H, "<span class='warning'>You cannot find any pheromones nearby.</span>")
+				return
+			// These are not ours, do not touch them
+			if(!(pheromones_to_destroy in active_pheromones_current))
+				to_chat(H, "<span class='warning'>These pheromones were created by someone else, you are unable to dissipate them.</span>")
+				return
+			// These are ours and we now destroy them
+			if(do_after(H, 3 SECONDS, FALSE, pheromones_to_destroy))
+				// Log the action
+				H.create_log(MISC_LOG, "destroyed pheromones that had the message of \"[pheromones_to_destroy.encoded_message]\"")
+
+				// Destroy it; the pheromones remove themselves from our list via signals
+				to_chat(H, "<span class='notice'>You dissipate your old pheromones.</span>")
+				qdel(pheromones_to_destroy)
+
+		// We decide to produce new ones
+		if("Produce")
+			// Can we create more pheromones?
+			if(length(active_pheromones_current) >= active_pheromones_maximum)
+				to_chat(H, "<span class='warning'>You already have [length(active_pheromones_current)] sets of pheromones active and are unable to produce any more.</span>")
+				return
+
+			// Encode the message
+			var/message_to_encode = input(H, "What message do you wish to encode? (max. [maximum_message_length] characters) Leave it empty to cancel.", "Produce Pheromones")
```

Comment:
Same here, I think you might be able to get the result as null optionally to let people cancel it.